packages/engine/Source/Scene/computeFlyToLocationForRectangle.js

```diff
@@ -53,7 +53,7 @@ function computeFlyToLocationForRectangle(rectangle, scene) {
           currentMax,
           item
         ) {
-          return Math.max(item.height, currentMax);
+          return Math.max(item.height, currentMax) || 0;
```

Comment:
Won't the value of currentMax be undefined or NaN?

Reply:
Based on [the documentation for `reduce`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce), the original value of `currentMax` is initiated at`-Number.MAX_VALUE`.

Also, we should avoid running the `Math.max` operation at all if we know it will return `NaN`. 

Although, since we end up adding the result to `positionWithoutTerrain`, it may make sense to confirm we found at least one valid height value before adding the accumulated value to `maxHeight`.

```js
        let heightFound = false;
        const maxHeight = positionsOnTerrain.reduce(function (
          currentMax,
          item
        ) {
         if (!defined(item.height)) {
              return currentMax;
          }
          heightFound = true;
          return Math.max(item.height, currentMax);
        },
        -Number.MAX_VALUE);

        const finalPosition = positionWithoutTerrain;
        if (heightFound) {
          finalPosition.height += maxHeight;
        }
        return finalPosition;

```

Reply:
ok, i think it works.